ape_safe/accounts.py

```diff
@@ -71,9 +72,28 @@ def account_file(self) -> dict:
     def address(self) -> AddressType:
         return self.network_manager.ethereum.decode_address(self.account_file["address"])
 
-    @property
+    @cached_property
     def contract(self) -> ContractInstance:
-        return self.chain_manager.contracts.instance_at(self.address)
+        safe_contract = self.chain_manager.contracts.instance_at(self.address)
+        if self.fallback_handler is None:
+            return safe_contract
+        else:
+            safe_abi = safe_contract.contract_type.dict()["abi"]
+            fallback_abi = self.fallback_handler.contract_type.dict()["abi"]
+            contract_type = ContractType.parse_obj(
+                {"contractName": safe_contract.contract_type.name, "abi": safe_abi + fallback_abi}
+            )
+            return ContractInstance(self.address, contract_type)
```

Comment:
if you do `self.chain_manager.contracts.instance_at(self.address, contract_type=contract_type)`, it would override the contract type with the new one so you wouldnt have to do this each time.